test/Events.IntegrationTest/Controllers/CollectControllerTests.cs

```diff
@@ -1,21 +1,63 @@
-﻿using System.Net.Http.Json;
+﻿using System.Net;
+using System.Net.Http.Json;
+using Bit.Core.AdminConsole.Entities;
+using Bit.Core.Entities;
 using Bit.Core.Enums;
+using Bit.Core.Repositories;
+using Bit.Core.Vault.Entities;
+using Bit.Core.Vault.Enums;
+using Bit.Core.Vault.Repositories;
 using Bit.Events.Models;
 
 namespace Bit.Events.IntegrationTest.Controllers;
 
-public class CollectControllerTests
+public class CollectControllerTests : IAsyncLifetime
 {
-    // This is a very simple test, and should be updated to assert more things, but for now
-    // it ensures that the events startup doesn't throw any errors with fairly basic configuration.
+    private EventsApplicationFactory _factory = null!;
+    private HttpClient _client = null!;
+    private string _ownerEmail = null!;
+    private Guid _ownerId;
+
+    public async Task InitializeAsync()
+    {
+        _factory = new EventsApplicationFactory();
+        _ownerEmail = $"integration-test+{Guid.NewGuid()}@bitwarden.com";
+        var (accessToken, _) = await _factory.LoginWithNewAccount(_ownerEmail);
+        _client = _factory.CreateAuthedClient(accessToken);
+
+        // Get the user ID
+        var userRepository = _factory.GetService<IUserRepository>();
+        var user = await userRepository.GetByEmailAsync(_ownerEmail);
+        _ownerId = user!.Id;
+    }
+
+    public Task DisposeAsync()
+    {
+        _client?.Dispose();
+        _factory?.Dispose();
+        return Task.CompletedTask;
+    }
+
     [Fact]
-    public async Task Post_Works()
+    public async Task Post_NullModel_ReturnsBadRequest()
     {
-        var eventsApplicationFactory = new EventsApplicationFactory();
-        var (accessToken, _) = await eventsApplicationFactory.LoginWithNewAccount();
-        var client = eventsApplicationFactory.CreateAuthedClient(accessToken);
+        var response = await _client.PostAsJsonAsync<IEnumerable<EventModel>?>("collect", null);
```

Comment:
Non-blockers: Consider using the Arrange, Act, and Assert pattern for the tests.